### ISSUE TITLE
Table "MSysRibbons" is not exported

### ISSUE THREAD

**Issue by Tanarri** [joyfullservice/msaccess-vcs-integration#207]
Hi,

I have exported a really big access database, which has saved global ribbons in table "USysRibbons".
After a full export the table "USysRibbons" wasn't exported to the directory "tables"

Cause is, that in my access database the table "USysRibbons" is named "usysribbons" (lowercase)
The table was manually created by me year ago when the access database was updated from access 2007 to 2013 to use the new features of access 2013

Solution:
Set clsOptions.TablesToExportData.CompareMode to vbTextCompare

I will prepare a pull request

**Comment by joyfullservice** [comment]
Thank you, @Tanarri! That is an excellent enhancement. Please feel free to close out this issue if the problem has been resolved. Thanks again for your contributions!

**Comment by Tanarri** [comment]
Fix doesn't work
Reason: When the DB options are loaded from "vcs-options.json" the property "Options.TablesToExportData" will be overwritten with a new dictionary object from the class "modJsonConverter.bas" and the setting "CompareMode = vbTextCompare" get lost.
So the patch #208 isn't enough.

I think we cannot patch the JSON Loader to use "CompareMode = vbTextCompare". This has to be tested well before.

**Comment by joyfullservice** [comment]
Let me take a closer look at this...

**Comment by joyfullservice** [comment]
I just pushed an update that implements `.CompareMode = TextCompare` across several different places where the list of table names is involved. This also ensures that the `CompareMode` is preserved when sorting a dictionary by keys. Let me know if this resolves the issue for you!

**Comment by Tanarri** [comment]
Hi Adam,

your commit #6688cd8 resolved the issue and the table "usysribbons" will be exported.  Great!

Does it make sense that we also ignone the case-sensitiv on queries, modules, reports.... ? I think so.
But if we go further and pull the files to git oder svn there there could be a conflict if developer A have a module "test.bas" and developer B has named a file "Test.bas"... Even more if the git server is linux, which can handle one file with different cases.

I don't have a solution for this, but I could be wrong - perhaps git  have a option to solve such a issue.

Greetings
Tanarri

**Comment by hecon5** [comment]
Tanarri, I've run across this a bit with git; I just tell it to be [case insensitive](https://stackoverflow.com/questions/8904327/case-sensitivity-in-git) and then any files that have to be renamed to the correct case can be handled on an as-needed basis. 

Personally, I'm of the mind that having two files with nearly the same names will lead to heartbreak at some point, especially with a vcs like this where 99% of the file handling is automated.

**Comment by Tanarri** [comment]
@hecon : Great if there is a solution for git to be case insensitive. This should be set if somebody uses vcs for access

@joyfullservice : I think it would be helpful if we implement `.CompareMode = TextCompare` for other access object like queries, modules, classes on export and import. What do you think?

**Comment by joyfullservice** [comment]
The reason the case sensitive aspect was important for the tables (as illustrated here by `USysRibbons`) was because VCS maintains a list of the _table names_ for exporting _data_. Adjusting for case-insensitive comparison makes sense here because this system table is treated special by Microsoft Access based on the _case-insensitive table name_. Furthermore, you cannot create two objects with the same name but different case. I cannot think of any negative side-affects for allowing case-insensitive _table data export_ as we have implemented in 6688cd8. However, we don't currently store a shared list of object names like this for other types of objects like queries, modules, classes, etc...

In my perspective, the purpose of this add-in is to export database objects as source code files, and to import/build database objects _from_ source code files. The file names are dynamically determined by the object names. If you rename `MyTable` to `mytable`, the add-in treats this as an entirely new and different object. On export, the old source file is purged out, and the renamed object exported using the new name. The index treats this file as a different object. This is the way it should work, since the file needs to reflect the new name when it is later imported back into a database. Git may (at its own discretion) determine that this was simply a rename, but that would be up to the VCS platform to make that determination.

In my opinion, I personally don't think the add-in needs to support two developers working on the same object with different names (casing difference). They should ideally get together and decide which name they want to use, and use that name going forward. To do otherwise just seems like you are opening yourself up for a lot of headaches and confusion down the road, not to mention adding complexity to the codebase.

That being said, this is just my perspective, having never encountered a situation where supporting different case names was needed. Is there a specific scenario (other than table data) that you could describe where this would be helpful?

**Comment by Tanarri** [comment]
Thank you for the very detailed description. After reading this I am of the same opinion as you. It doesn't make sense to make the codebase more complex, since there shouldn't really be any problems.

My idea comes from the fact that years ago I myself created an Access project that exported all Access objects to put them into an SVN repository. I still do this and have already made about 4000 commits in it. I used the SVN Repos alone and not across different PCs or developers. It works well.
At the beginning of my project I had problems with case sensitivity of some queries or modules, because I had named them inconsistently. Gradually I had renamed them to lower case and encountered problems with the Linux SVN server. My solution at that time was to run a check routine before exporting the access objects, which checked if all access objects are lowercase. Thus only objects were exported, which were completely written in small letters. This solved my problems.
A solution could also have been that I would have had to delete an object that was renamed and recreate it (as it is done here).

We can close this issue.

Translated with www.DeepL.com/Translator (free version)
